[Bug]: Runtime Error (SIGSEGV) with the terraform provider for Grafana

### Terraform Version

v1.5.0

### Terraform Grafana Provider Version

v3.5.0

### Grafana Version

v11.2.0-73451

### Affected Resource(s)

The ability to create alert rule groups via terraform.

### Terraform Configuration Files

_No response_

### Expected Behavior

Expected the TF plan to run and show me the changes to add alerts.

### Actual Behavior

Got the following error when running the TF plan:

Stack trace from the terraform-provider-grafana_v3.5.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x147d515]

goroutine 63 [running]:
github.com/grafana/terraform-provider-grafana/v3/internal/resources/grafana.datasourceDatasourceRead({0x4a49ed?, 0x1dd0fa8?}, 0xc00077e680, {0x17b51a0?, 0xc0001425a0?})
github.com/grafana/terraform-provider-grafana/v3/internal/resources/grafana/data_source_data_source.go:53 +0x155
github.com/grafana/terraform-provider-grafana/v3/internal/resources/grafana.addValidationToSchema.func1({0x1dd1050?, 0xc0005d7ce0?}, 0x1176592e000?, {0x17b51a0?, 0xc0001425a0?})
github.com/grafana/terraform-provider-grafana/v3/internal/resources/grafana/resources.go:41 +0x10b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00070eee0, {0x1dd0fa8, 0xc000968690}, 0xc00077e680, {0x17b51a0, 0xc0001425a0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:823 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc00070eee0, {0x1dd0fa8, 0xc000968690}, 0xc00077e580, {0x17b51a0, 0xc0001425a0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:1043 +0x13a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000610348, {0x1dd0fa8?, 0xc0009685d0?}, 0xc0009682a0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1434 +0x6b1
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadDataSource(0xc0005718f0, {0x1dd0fa8?, 0xc000968300?}, 0xc0009682a0)
github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ReadDataSource.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc000117540, {0x1dd0fa8?, 0xc000956f90?}, 0xc0000b1c70)
github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:688 +0x27d
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x1a51800, 0xc000117540}, {0x1dd0fa8, 0xc000956f90}, 0xc00077e080, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:572 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0006ba000, {0x1dd0fa8, 0xc000956f00}, {0x1dda780, 0xc0004ed680}, 0xc00081cea0, 0xc000668c90, 0x2dc31c0, 0x0)
google.golang.org/grpc@v1.65.0/server.go:1379 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc0006ba000, {0x1dda780, 0xc0004ed680}, 0xc00081cea0)
google.golang.org/grpc@v1.65.0/server.go:1790 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
google.golang.org/grpc@v1.65.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 10
google.golang.org/grpc@v1.65.0/server.go:1040 +0x125

Error: The terraform-provider-grafana_v3.5.0 plugin crashed!

### Steps to Reproduce

Terraform plan

### Important Factoids

_No response_

### References

_No response_

--- a/internal/resources/grafana/data_source_data_source.go
+++ b/internal/resources/grafana/data_source_data_source.go
@@ -44,10 +44,16 @@ func datasourceDatasourceRead(ctx context.Context, d *schema.ResourceData, meta 
 		resp, err = client.Datasources.GetDataSourceByName(name.(string))
 	} else if uid, ok := d.GetOk("uid"); ok {
 		resp, err = client.Datasources.GetDataSourceByUID(uid.(string))
+	} else {
+		return diag.Errorf("name or uid must be set")
 	}
 
 	if err != nil {
 		return diag.FromErr(err)
+	}
+
+	if resp == nil {
+		return diag.Errorf("unexpected state, API response is nil")
 	}
 
 	return datasourceToState(d, resp.GetPayload())

